src/test/java/com/artipie/maven/MavenProxyIT.java

```diff
@@ -0,0 +1,172 @@
+ * SOFTWARE.
+ */
+package com.artipie.maven;
+
+import com.amihaiemil.eoyaml.Yaml;
+import com.artipie.ArtipieServer;
+import com.artipie.docker.junit.DockerClientSupport;
+import com.jcabi.log.Logger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import org.cactoos.list.ListOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.AllOf;
+import org.hamcrest.core.IsNot;
+import org.hamcrest.core.StringContains;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.io.TempDir;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+
+/**
+ * Integration test for {@link com.artipie.maven.http.MavenProxySlice}.
+ *
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @since 0.11
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+@EnabledOnOs(OS.LINUX)
+@DockerClientSupport
+final class MavenProxyIT {
+
+    /**
+     * Temporary directory for all tests.
+     * @checkstyle VisibilityModifierCheck (3 lines)
+     */
+    @TempDir
+    Path tmp;
+
+    /**
+     * Subdirectory in temporary directory.
+     */
+    private Path subdir;
+
+    /**
+     * Local Artipie server that stores artifacts.
+     */
+    private ArtipieServer server;
+
+    /**
+     * Container for local server.
+     */
+    private GenericContainer<?> cntn;
+
+    /**
+     * Proxy Artipie server port.
+     */
+    private int port;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        this.subdir = Files.createDirectory(Path.of(this.tmp.toString(), "subdir"));
+        this.server = new ArtipieServer(this.subdir, "my-maven", this.configsProxy());
+        this.port = this.server.start();
+        final Path setting = this.subdir.resolve("settings.xml");
+        setting.toFile().createNewFile();
+        Files.write(setting, this.settings());
+        this.cntn = new GenericContainer<>("centos:centos8")
+            .withCommand("tail", "-f", "/dev/null")
+            .withWorkingDirectory("/home/")
+            .withFileSystemBind(this.subdir.toString(), "/home");
+        Testcontainers.exposeHostPorts(this.port);
+        this.cntn.start();
+        this.cntn.execInContainer("yum", "-y", "install", "maven");
+    }
+
+    @AfterEach
+    void tearDown() {
+        this.server.stop();
+        this.cntn.close();
+    }
+
+    @Test
+    void shouldGetArtifactFromLocalAndSaveInCache() throws Exception {
+        final String artifact = "-Dartifact=aspectj:aspectj-ant:1.0.6:jar";
+        this.exec("mvn", "-s", "/home/settings.xml", "dependency:get", artifact);
+        MatcherAssert.assertThat(
+            this.exec(
+                "mvn", "-s", "/home/settings.xml", "dependency:get", artifact
```

Comment:
@genryxy I see that this command is executed twice, is it really necessary?

Reply:
@olenagerasimova thanks. I try to verify that artifacts are cached. Maybe, it would be nice just to check that storage contains the key, but I have some troubles with it therefore implemented so. I'll try to check through storage and execute this command only once.

Reply:
@genryxy have you tried simply check temp directory structure? You can add brake point, copy path and use `ls -la`. As far as see files should be `./subdir/repos/`. When you understand where files are, you can create `FileStorage` instance over subdir and check that files are loaded by corresponding keys

Reply:
@olenagerasimova yes, I tried to check temp directory. It has `artipie.yaml`, `settings`, `repos/my-maven.yaml`, `repos/_credentials.yml`